enterprise/server/raft/replica/replica.go

```diff
@@ -166,16 +167,49 @@ func (sm *Replica) Usage() (*rfpb.ReplicaUsage, error) {
 	iter := db.NewIter(iterOpts)
 	defer iter.Close()
 
-	estimatedBytesUsed := int64(0)
 	for iter.First(); iter.Valid(); iter.Next() {
-		sizeBytes, err := sizeOf(iter.Key(), iter.Value())
+		partID, sizeBytes, err := sizeOf(iter.Key(), iter.Value())
 		if err != nil {
-			return nil, err
+			return nil, nil, err
+		}
+		pu := usageByPartition[partID]
+		if pu == nil {
+			pu = &rfpb.PartitionMetadata{PartitionId: partID}
+			usageByPartition[partID] = pu
+		}
+		pu.SizeBytes += sizeBytes
+		if bytes.Contains(iter.Key(), casDir) {
+			pu.CasCount += 1
+		} else if bytes.Contains(iter.Key(), acDir) {
+			pu.AcCount += 1
 		}
-		estimatedBytesUsed += sizeBytes
 	}
-	ru.EstimatedDiskBytesUsed = estimatedBytesUsed
-	return ru, nil
+
+	var partitionUsages []*rfpb.PartitionMetadata
+	for _, u := range usageByPartition {
+		partitionUsages = append(partitionUsages, u)
+	}
```

Comment:
Not against doing it this way, but it feels like it might be inefficient to scan the whole range like this (lots of proto deserialization etc). I was thinking we could use this: https://pkg.go.dev/github.com/cockroachdb/pebble#DB.EstimateDiskUsage, and if we tracked the set of active partitions in a region we could just call that once for each region?